Sources/SwiftGRPCNIO/GRPCWebCorsHandler.swift

```diff
@@ -0,0 +1,73 @@
+import NIO
+import NIOHTTP1
+
+/// Handler that manages the CORS protocol for requests incoming from the browser.
+class GRPCWebCorsHandler {
+  var requestMethod: HTTPMethod?
+}
+
+extension GRPCWebCorsHandler: ChannelInboundHandler {
+  typealias InboundIn = HTTPServerRequestPart
+  typealias OutboundOut = HTTPServerResponsePart
+
+  func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+    // If the request is OPTIONS, the request is not propagated further.
+    switch self.unwrapInboundIn(data) {
+    case .head(let requestHead):
+      requestMethod = requestHead.method
+      if requestMethod == .OPTIONS {
+        var headers = HTTPHeaders()
+        headers.add(name: "Access-Control-Allow-Origin", value: "*")
+        headers.add(name: "Access-Control-Allow-Methods", value: "POST")
+        headers.add(name: "Access-Control-Allow-Headers",
+                    value: "content-type,x-grpc-web,x-user-agent")
+        headers.add(name: "Access-Control-Max-Age", value: "86400")
+        ctx.write(self.wrapOutboundOut(.head(HTTPResponseHead(version: requestHead.version,
+                                                              status: .ok,
+                                                              headers: headers))),
+                  promise: nil)
+        return
+      }
+    case .body(_):
```

Comment:
super-nit: newline above this line, and (optionally) omit `(_)`?

Reply:
done